### PR TITLE
L1TP2: Removing the old PV Simulation: TkPrimaryVertex

### PR DESCRIPTION
**Pull request by BenjaminRS**
#### PR description:

This PR updates the Phase 2 code to remove older Primary Vertex simulation: TkPrimaryVertex. A number of files had to be touched in order to update it to use the latest PV emulation. N.B.: the L1TkEmParticleProducer class needs to be updated by Gianluca.

#### PR validation:

I have tested with the L1Trigger/L1TTrackMatch/test/L1TrackObjectNtupleMaker_cfg.py which works. However with the following command:
`cmsDriver.py step1\
 --conditions 123X_mcRun4_realistic_v3 -n 10\
 --era Phase2C9\
 --eventcontent FEVTDEBUGHLT\
 --step RAW2DIGI,L1TrackTrigger,L1\
 --customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_1000,L1Trigger/Configuration/customisePhase2TTNoMC.customisePhase2TTNoMC,Configuration/DataProcessing/Utils.addMonitoring\
 --geometry Extended2026D49\
 --no_exec\
 --nThreads 8\
 --python runL1.py\
 --filein das://store/mc/Phase2HLTTDRSummer20ReRECOMiniAOD/SMS-T2tt_mStop-1000_mLSP-775_TuneCP2_14TeV-madgraph-pythia8/FEVT/PU200_withNewMB_111X_mcRun4_realistic_T15_v1-v1/30000/044ED959-E2AF-4F43-9BB2-99A5CE18C2E9.root\
 --processName=L1REPR`
I get an error:

> An exception of category 'DictionaryNotFound' occurred while
>    [0] Constructing the EventProcessor
>    [1] Calling ProductRegistry::initializeLookupTables, checking dictionaries for consumed products
> Exception Message:
> No data dictionary found for the following classes:
> 
>   edm::Wrapper<std::vector<l1t::TkPrimaryVertex> >
>   std::vector<l1t::TkPrimaryVertex>

Which I am not sure how to fix

### CODE REVIEW

**Comment by BenjaminRS** [comment]
Notifying @cerminar for the L1TkEmParticleProducer

File: L1Trigger/VertexFinder/python/VertexProducer_cff.py [hunk 1]
```diff
@@ -53,13 +53,13 @@
...
         # Maximum pt of tracks used to create vertex [GeV]
-        VxMaxTrackPt = cms.double(50.0),
+        VxMaxTrackPt = cms.double(127.0),
         # When the track pt > VxMaxTrackPt, how should the tracks be considered
         #   -1 = tracks are valid
         #   0 = tracks are mismeasured and ignored/truncated
         #   1 = tracks are mismeasured and saturate at VxMaxTrackPt
         # Option '0' was used for the TDR, but '1' is used for the firmware
-        VxMaxTrackPtBehavior = cms.int32(0),
+        VxMaxTrackPtBehavior = cms.int32(1),
```
Comment (BenjaminRS) [inline L1Trigger/VertexFinder/python/VertexProducer_cff.py:62]: Also took the opportunity to update the default values to match the FW and fix a bug reported [here](https://indico.cern.ch/event/1139008/#10-tracks-vertex)

**Comment by cbotta** [comment]
@BenjaminRS. Q1: what about the HPS Taus. They also had this dependency on the old vtx, what's the plan there? Q2: about the bug-fix, I don't remember, does it only affect the emulation for high pT samples? (given it degrades the emulation performance I would like to be sure we know which is the expected effect) Thanks!

**Comment by BenjaminRS** [comment]
A1. Yes all the HPSTaus needs to be updated. In particular the following files:
> DataFormats/L1TParticleFlow/interface/HPSPFTau.h
> DataFormats/L1TParticleFlow/src/HPSPFTau.cc
> L1Trigger/Phase2L1Taus/interface/L1HPSPFTauBuilder.h
> L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc
> L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.h
> L1Trigger/Phase2L1Taus/python/l1emulator_cff.py
> L1Trigger/Phase2L1Taus/src/L1HPSPFTauBuilder.cc
> L1Trigger/Phase2L1Taus/test/produceHPSPFTaus_cfg.py
I understood that Sandeep was taking care of that.

A2: It only affects FastHisto Simulation of the high pT samples (like from a 6TeV Z'). This makes the simulation match the FastHisto Emulation.

**Comment by BenjaminRS** [comment]
@sandeepbhowmik1 How is it going with updating the HPS Taus?

**Comment by cerminar** [comment]
> L1TkEmParticleProducer



> Notifying @cerminar for the L1TkEmParticleProducer

Thanks!. The plugin is not needed anymore and it is among those removed by PR https://github.com/cms-l1t-offline/cmssw/pull/1027

**Comment by BenjaminRS** [comment]
Sandeep has made the PR to update the HPSTaus with: https://github.com/cms-l1t-offline/cmssw/pull/1030

**Comment by cecilecaillol** [comment]
tagged as l1t-phase2-v3.4.36

**Pull request merged**